src/DependencyManagement/DependencyManagerStorage.cs

```diff
@@ -90,5 +90,32 @@ public DateTime GetSnapshotCreationTimeUtc(string path)
         {
             return Directory.GetCreationTimeUtc(path);
         }
+
+        public void SetSnapshotAccessTimeToUtcNow(string path)
+        {
+            var markerFilePath = DependencySnapshotFolderNameTools.CreateLastAccessMarkerFilePath(path);
+            if (File.Exists(markerFilePath))
+            {
+                File.SetLastWriteTimeUtc(markerFilePath, DateTime.UtcNow);
+            }
+            else
+            {
+                var file = File.Open(markerFilePath, FileMode.Create, FileAccess.Write, FileShare.ReadWrite | FileShare.Delete);
```

Comment:
Should we use a ```using statement``` to ensure that the file is close in case of an exception occurs while doing File.Open(..)?

Reply:
If `File.Open` throws, it does not return anything, so `using` will not be able to call `.Dispose` anyway, and the behavior will be exactly the same. This is just a matter of taste: I found that `using` with an empty body invites more questions than just explicitly invoking `Dispose`.

Reply:
Sounds good. We'll keep it as is.